### DISCUSSION TITLE
Get multiple image annotations with one request?

### DISCUSSION THREAD

**Discussion by ymelnikova** [biigle/core#976] [Q&A]
Hi team, 
Is there a way to get annotations from images in bulk via one http request?
Currently, it is a single request per image and it takes 8 seconds for around 10 images. I looked at using threads and asynchronous calls with aiohttp,  which is a working option. 
But maybe something with Biigle API is possible, or can become a future feature?

**Comment by mzur** [comment]
Currently there is no API endpoint that allows you to fetch many annotations directly. As a workaround, you could request a report (CSV or COCO). The request will return the report ID and you can poll BIIGLE for the finished report. Once finished, you can download the file and extract the annotation information.
